src/App.tsx

```diff
@@ -1,26 +1,43 @@
-import React, { Component } from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { FindMovie } from './components/FindMovie';
 import data from './api/movies.json';
+import { MoviesCard } from './components/interfaces';
 
-export class App extends Component {
-  state = {
-    movies: data,
+export const App: React.FC = () => {
+  const [hasAlready, setHasAlreadyStatus] = useState(false);
+  const [movies, setMovies] = useState(data);
+
+  const isNotHasAlready = () => {
+    setHasAlreadyStatus(false);
   };
 
-  render() {
-    const { movies } = this.state;
+  const addFilm = (newFilm: MoviesCard) => {
+    const isHasAlready = movies
+      .some(movie => movie.imdbId === newFilm.imdbId);
+
+    if (!isHasAlready) {
+      const dataWithNewMovies = [...movies, newFilm];
 
-    return (
-      <div className="page">
-        <div className="page-content">
-          <MoviesList movies={movies} />
-        </div>
-        <div className="sidebar">
-          <FindMovie />
-        </div>
+      setMovies(dataWithNewMovies);
+    } else {
+      setHasAlreadyStatus(true);
+    }
+  };
+
+  return (
+    <div className="page">
+      <div className="page-content">
+        <MoviesList movies={movies} />
```

Comment:
```suggestion
        <Movies data={movies} />
```
можна ще так, але можна посперечатися бо назва `data` дуже загальна, будь-що може бути як data

Reply:
Тут тоже не я)))
https://prnt.sc/sgyfp2